internal/backends/winit/renderer/skia/textlayout.rs

```diff
@@ -46,6 +46,14 @@ pub fn create_layout(
     h_align: items::TextHorizontalAlignment,
     overflow: items::TextOverflow,
 ) -> skia_safe::textlayout::Paragraph {
+    // For the Qt backend, text_size() with max_width == Some(0.) results in an empty
+    // bounding rect and Qt::Qt::TextWordWrap being omitted from the text flags. That
```

Comment:
```suggestion
    // bounding rect and Qt::TextWordWrap being omitted from the text flags. That
```

I don't understand what the qt backend has anything to do with skia.

What does the femtovg renderer do? And the software renderer? 
what's the problem here? that the minimum height is set to a high value because the function is called with zero width?   this sounds like a problem in the layout engine

What would happen with such an example:
```
export App := Window {
preferred-width: 300px;
preferred-height: 300px;

VerticalLayout {
    padding: -10px;
    Text {
        text: "Lots of text with\nnewlines etc.";
        wrap: word-wrap;
    }
}
}
```

because of the negative margin, then the max width will not be zero,  but still too small, wouldn't it?

Reply:
> I don't understand what the qt backend has anything to do with skia.

This is only about the alignment of behavior - short of changing other mechanisms such as window item width/height defaults, it seems most sensible to me to make a change here and explain this behavior (Some(0) means None) by alignment with Qt.

> What does the femtovg renderer do? 

I outlined that in the second point in the commit message.

>And the software renderer?

The text layout used in the software renderer behaves like Skia (the window gets way to tall, but then after a while the renderer asserts with an out-of-slice panic). Once we decided on a path for this, I was going to change that as well (with a unit test).

> what's the problem here? that the minimum height is set to a high value because the function is called with zero width?

Yes. Laying out text with a constrained width of 0 basically creates one line per glyph.

> this sounds like a problem in the layout engine

The text layout or the item layouts?

> What would happen with such an example:

This still breaks indeed. With Qt it "works" because the padding is combined with the width of the window item, which is 640 only for Qt. But if I comment out that part in qt_window then it also breaks with Qt and the overall window height is broken.

Good example. This makes me rather wonder if instead we have the following logic in both backends:

```rust
fn show(&self) {
    // ...
    let window_item = ...;

    // Avoid computing the layout info with non-sensical sizes. 640x480 is also the size
    // of a top-level QWidget that has never been shown.
    if window_item.width().approx_eq(&0.) {
        window_item.width.set(640);
    }
    if window_item.height().approx_eq(&0.) {
        window_item.height.set(480);
    }

    let layout_info_h = component.as_ref().layout_info(Orientation::Horizontal);
    let layout_info_v = component.as_ref().layout_info(Orientation::Vertical);
    // ...
}
```